go-controller/pkg/node/controllers/egressip/egressip.go

```diff
@@ -654,9 +654,10 @@ func (c *Controller) GetLinkContainingIPAndValidate(ip net.IP, expectedNetwork s
 	if !found {
 		return false, nil, nil
 	}
-	if network.String() != expectedNetwork {
-		return false, nil, fmt.Errorf("EgressIP %s is was assigned to network %s but we got network %s. Mismatch therefore not assigning",
-			ip.String(), expectedNetwork, network.String())
+	// network may not have been assigned yet and we don't depend on status
+	if expectedNetwork != "" && network.String() != expectedNetwork {
+		klog.Errorf("EgressIP %s is assigned to network %q by cluster manager but EIP controller calculated the network "+
```

Comment:
So I guess this is never expected. Cluster manager assigned some IP based on host addresses, and now the network doesnt match on the node? Couldn't that be because someone took the interface down or removed the IP address? Just trying to understand if this error always signifies that there is a bug in our code, or if something else could cause it.

Reply:
yeah - it would be unexpected and unlikely to occur.

It could occur due to race condition between node ip manager and CM as you describe or it could occur due to a bug in our code. 

I wanted to highlight it and warning us via the logs. Ill extract this check from this func. Func is doing 2 things.